src/modules/csm/checkpoint.py

```diff
@@ -229,18 +234,48 @@ def _prepare_committees(self, epoch: EpochNumber) -> Committees:
 
 def process_attestations(attestations: Iterable[BlockAttestation], committees: Committees) -> None:
     for attestation in attestations:
-        committee_id = (attestation.data.slot, attestation.data.index)
-        committee = committees.get(committee_id, [])
-        att_bits = _to_bits(attestation.aggregation_bits)
-        for index_in_committee, validator_duty in enumerate(committee):
-            validator_duty.included = validator_duty.included or _is_attested(att_bits, index_in_committee)
+        committee_offset = 0
+        for committee_idx in get_committee_indices(attestation):
+            committee = committees.get((attestation.data.slot, committee_idx), [])
+            att_bits = hex_bitlist_to_list(attestation.aggregation_bits)[committee_offset:][: len(committee)]
+            for index_in_committee in get_set_indices(att_bits):
+                committee[index_in_committee].included = True
+            committee_offset += len(committee)
+
+
+def get_committee_indices(attestation: BlockAttestation) -> list[CommitteeIndex]:
+    if is_eip7549_attestation(attestation):
+        return [str(i) for i in get_set_indices(hex_bitvector_to_list(attestation.committee_bits))]
+    return [attestation.data.index]
+
+
+def is_eip7549_attestation(attestation: BlockAttestation) -> TypeGuard[BlockAttestationElectra]:
+    # @see https://eips.ethereum.org/EIPS/eip-7549
+    has_committee_bits = getattr(attestation, "committee_bits") is not None
+    has_zero_index = attestation.data.index == "0"
+    if has_committee_bits and not has_zero_index:
+        raise ValueError(f"Got invalid {attestation=}")
+    return has_committee_bits and has_zero_index
+
+
+def get_set_indices(bits: Sequence[bool]) -> list[int]:
```

Comment:
https://github.com/ethereum/consensus-specs/blob/2c1f677187e6534aec77057a7d1cc746a40d3630/specs/electra/beacon-chain.md#get_committee_indices

Reply:
Add this link pls

Reply:
I think it's not exactly what it does. This function is more generic.